narwhals/_pandas_like/group_by.py

```diff
@@ -96,16 +102,16 @@ def _from_native_frame(self, df: PandasLikeDataFrame) -> PandasLikeDataFrame:
         )
 
     def __iter__(self) -> Iterator[tuple[Any, PandasLikeDataFrame]]:
-        with warnings.catch_warnings():
-            # we already use `tupleify` above, so we're already opting in to
-            # the new behaviour
-            warnings.filterwarnings(
-                "ignore",
-                message="In a future version of pandas, a length 1 tuple will be returned",
-                category=FutureWarning,
-            )
-            iterator = self._grouped.__iter__()
-        yield from ((key, self._from_native_frame(sub_df)) for (key, sub_df) in iterator)
+        indices = self._grouped.indices
+        for key in indices:
+            if (
+                self._df._implementation is Implementation.PANDAS
+                and self._df._backend_version < (2, 2)
+            ):  # pragma: no cover
+                pass
+            else:  # pragma: no cover
+                key = tupleify(key)  # noqa: PLW2901
+            yield (key, self._from_native_frame(self._grouped.get_group(key)))
```

Comment:
I am a bit confused by this path, wouldn't `self._grouped.get_group(key)` break after converting to a tuple if that's not the native behavior?

Reply:
it's to deal with pandas' deprecation warning:
```python
In [12]: pd.DataFrame({'a': [1,2,3], 'b': [4,5,6]}).groupby(['a']).get_group(1)
<ipython-input-12-3b83f23f8427>:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  pd.DataFrame({'a': [1,2,3], 'b': [4,5,6]}).groupby(['a']).get_group(1)
Out[12]:
   a  b
0  1  4

In [13]: pd.DataFrame({'a': [1,2,3], 'b': [4,5,6]}).groupby(['a']).get_group((1,))
Out[13]:
   a  b
0  1  4
```

Reply:
Got it! We deal with returning a tuple of `(tuple of keys, dataframe)` in the upstream `GroupBy` class anyway! Sorry for my confusion